### PR TITLE
move utilities from drivers/misc to lib/

### PR DESCRIPTION
**Pull request by nashif**
Move things from misc/ and drivers/ to lib/. This should be a place for all
kind of code that can be used by different subsystems.

### CODE REVIEW

**Comment by galak** [comment]
I think crc, ring_buff, should just live in lib/

**Comment by nashif** [comment]
> I think crc, ring_buff, should just live in lib/

yeah, sounds good

**Review by dbkinder — COMMENTED** [review]
doc changes look OK, but there's a build failure that might be masking something...

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
-add_subdirectory(libc)
+add_subdirectory(crc)
```
Comment (galak) [inline lib/CMakeLists.txt]: Do we really need a subdir crc, how about just having it be lib/crc16_sw.c

**Review by galak — COMMENTED** [review]
Similar comment about need dir for ring buf as for crc.  Seems like we can just have lib/crc16_sw.c and lib/ring_buffer.c

**Comment by nashif** [comment]
> Similar comment about need dir for ring buf as for crc. Seems like we can just have lib/crc16_sw.c and lib/ring_buffer.c

this will get really crowded under lib/, directories are cheap, we already have a few other crc files coming and we will certainly have things in directories in the future, so why not directories.

File: drivers/console/ipm_console_receiver.c [hunk 2]
```diff
@@ -9,7 +9,7 @@
 #include <errno.h>
 
 #include <kernel.h>
-#include <misc/ring_buffer.h>
```
Comment (andrewboie) [inline drivers/console/ipm_console_receiver.c]: This is technically API breakage.
We alright with that?

**Review by tyrantlizard77 — APPROVED** [review]

File: drivers/console/ipm_console_receiver.c (hunk 2 above)
Comment (nashif) [reply drivers/console/ipm_console_receiver.c]: addressed

**Pull request merged**